sycl/doc/extensions/proposed/sycl_ext_oneapi_device_wait.asciidoc

```diff
@@ -0,0 +1,201 @@
+device to complete.
+This functionality is similar to the CUDA API `cudaDeviceSynchronize`.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_ONEAPI_DEVICE_WAIT` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== New aspect
+
+This extension adds the following aspect.
+
+[source,c++]
+----
+namespace sycl {
+
+enum class aspect {
+  // ...
+  ext_oneapi_device_wait
+};
+
+} // namespace sycl
+----
+
+'''
+
+[source,c++]
+----
+ext_oneapi_device_wait
+----
+
+Indicates that the device supports the member functions described below.
+
+'''
+
+=== New member functions for the device class
+
+This extension adds the following member functions to the `device` class.
+
+[source,c++]
+----
+namespace sycl {
+
+class device {
+  // ...
+  void ext_oneapi_wait();
+  void ext_oneapi_wait_and_throw();
+  void ext_oneapi_throw_asynchronous();
+};
+
+} // namespace sycl
+----
+
+'''
+
+[source,c++]
+----
+void ext_oneapi_wait();
+----
+
+_Effects:_ Blocks the calling thread until all commands previously submitted to
+any queue on this device have completed.
+
+_Throws:_ A synchronous `exception` with the `errc::feature_not_supported`
+error code if the device does not have `aspect::ext_oneapi_device_wait`.
+
+'''
+
+[source,c++]
+----
+void ext_oneapi_wait_and_throw();
+----
+
+_Effects:_ Blocks the calling thread until all commands previously submitted to
+any queue on this device have completed.
+
+At least all unconsumed asynchronous errors held by any queue (or its associated
+context) on this device are passed to the appropriate async_handler as described
+in section 4.13.1.3 "Priorities of async handlers" of the core SYCL
+specification.
+
+_Throws:_ A synchronous `exception` with the `errc::feature_not_supported`
+error code if the device does not have `aspect::ext_oneapi_device_wait`.
+
+'''
+
+[source,c++]
+----
+void ext_oneapi_throw_asynchronous();
+----
+
+_Effects:_ Checks to see if any unconsumed asynchronous errors have been
+produced by any queue (or its associated context) on this device.
+If so, they are passed to the appropriate async_handler as described in section
+4.13.1.3 "Priorities of async handlers" of the core SYCL specification.
+
+_Throws:_ A synchronous `exception` with the `errc::feature_not_supported`
+error code if the device does not have `aspect::ext_oneapi_device_wait`.
```

Comment:
@gmlueck - Sorry for the post-commit review, but is there a reason we would need this aspect for `ext_oneapi_throw_asynchronous()`?